test/services/query-builder.test.js

```diff
@@ -70,20 +71,82 @@ describe('services > query-builder', () => {
 
         expect(includes).toStrictEqual([{
           as: 'userAccessor',
-          attributes: ['uid', 'name'],
+          attributes: ['p_id', 'uid', 'name'],
           model: { name: 'user' },
         }]);
       });
 
-      it('should always include the target key even if not specified', () => {
+      it('should always include the source key and primary key even if not specified', () => {
         expect.assertions(1);
         const { model, queryBuilder } = setup();
 
         const includes = queryBuilder.getIncludes(model, ['user.name', 'user.id', 'user.badField']);
 
         expect(includes).toStrictEqual([{
           as: 'userAccessor',
-          attributes: ['uid', 'name'],
+          attributes: ['p_id', 'uid', 'name'],
+          model: { name: 'user' },
+        }]);
+      });
+    });
+
+    describe('with a belongsTo relationship', () => {
+      function setup() {
+        const target = {
+          primaryKeyAttributes: ['pid'],
+          tableAttributes: {
+            pid: { field: 'PID', fieldName: 'p_id' },
+            id: { field: 'Uid', fieldName: 'uid' },
+            name: { field: 'Name', fieldName: 'name' },
+          },
+          unscoped: () => ({ name: 'user' }),
+        };
+
+        const association = {
+          associationType: 'BelongsTo',
+          as: 'user',
+          associationAccessor: 'userAccessor',
+          target,
+          targetKey: 'id',
+        };
+
+        const model = {
+          name: 'address',
+          associations: [association],
+        };
+
+        const sequelizeOptions = { sequelize: Sequelize };
+        Interface.Schemas = { schemas: { actor: { idField: 'id' } } };
+
+        const queryBuilder = new QueryBuilder(model, sequelizeOptions, {});
+
+        return {
+          association, model, target, queryBuilder,
+        };
+      }
+
+      it('should exclude field names that do not exist on the table', async () => {
+        expect.assertions(1);
+        const { model, queryBuilder } = setup();
+
+        const includes = queryBuilder.getIncludes(model, ['user.uid', 'user.name', 'user.id', 'user.badField']);
+
+        expect(includes).toStrictEqual([{
+          as: 'userAccessor',
+          attributes: ['p_id', 'uid', 'name'],
+          model: { name: 'user' },
+        }]);
+      });
+
+      it('should always include the target key and primary key even if not specified', () => {
+        expect.assertions(1);
+        const { model, queryBuilder } = setup();
+
+        const includes = queryBuilder.getIncludes(model, ['user.name', 'user.id', 'user.badField']);
+
+        expect(includes).toStrictEqual([{
+          as: 'userAccessor',
+          attributes: ['p_id', 'uid', 'name'],
```

Comment:
Kind of same question as above ☝️ 

I don't get why the target `'uid'` is set here.
Don't we just need:
- the primary key of the record (for ember data)
- the displayed field for the data display in the user interface.
?

Reply:
Yes, I changed the behavior to do this instead of returning the source/target fields plus the primary key